app/javascript/components/CaseNoteForm/index.js

```diff
@@ -111,7 +111,14 @@ class CaseNoteForm extends React.Component {
         };
 
         apiPost('/api/case_notes', { case_note: body })
-          .then(() => window.location.reload())
+          .then(response => {
+            if (this.props.appendCaseNote) {
```

Comment:
CaseNote creation can happen in 1 of 2 components: `ParticipantCard` from the `StaffDashboard` page or `CaseNoteContainer` in the `ParticipantShowPage`. The `CaseNoteContainer` passes the `appendCaseNote` callback and the `ParticipantCard` passes the `incrementNumCaseNotes` callback. I could shorten this code (delete the if-else) to just always run a callback called `caseNoteCreationCallback` that always takes in `response.data` since only 1 of them is ever passed at a time but I didn't know if that was the right choice because I thought it was significantly less readable (especially since incrementNumCaseNotes doesn't care about the data). If you guys want just a single callback though, let me know and I can change it. Let me know if what I'm talking about isn't clear not sure if I articulated it correctly.

Reply:
This makes sense to me!